alembic/script/revision.py

```diff
@@ -40,6 +40,34 @@ def __init__(self, message, argument):
         self.argument = argument
 
 
+class LoopDetected(RevisionError):
+    def __init__(self, revision):
+        self.revision = revision
+        super(LoopDetected, self).__init__(
+            "Self-loop is detected on revision %s" % (revision,)
+        )
+
+
+class DependencyLoopDetected(RevisionError):
+    def __init__(self, revision):
+        self.revision = revision
+        super(DependencyLoopDetected, self).__init__(
+            "Dependency self-loop is detected on revision %s" % (revision,)
+        )
+
+
+class CycleDetected(RevisionError):
+    def __init__(self):
+        super(CycleDetected, self).__init__("Cycle is detected")
+
```

Comment:
**mike bayer** (zzzeek) wrote:

in SQLAlchemy where we also have a "topological" function we actually load up the "cycle" exception with all the involved members.   if possible that could be useful here since if someone encounters this error they may want to see what revs are invovled.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/2379